modules/tooltip/react/lib/OverflowTooltip.tsx

```diff
@@ -0,0 +1,133 @@
+import * as React from 'react';
+import innerText from 'react-innertext';
+
+import {getTransformFromPlacement, Placement, Popper} from '@workday/canvas-kit-react-popup';
+import {mergeCallback} from '@workday/canvas-kit-react-common';
+
+import {TooltipContainer} from './TooltipContainer';
+import {useTooltip} from './useTooltip';
+
+/**
+ * Look for an element within the tree for a `text-overflow` CSS property of `ellipsis`.
+ * This could be the passed element, or a descendant. If no element is found, `null` is returned.
+ */
+const findOverflowElement = (element: HTMLElement): HTMLElement | null => {
+  const style = getComputedStyle(element);
+  if (
+    style.overflow === 'auto' ||
+    style.overflow === 'scroll' ||
+    style.overflow === 'clip' ||
+    style.overflow === 'hidden'
+  ) {
+    return element;
+  } else {
+    for (let i = 0; i < element.children.length; i++) {
+      const overflowElement = findOverflowElement(element.children[i] as HTMLElement);
+      if (overflowElement) {
+        return overflowElement;
+      }
+    }
+    return null;
+  }
+};
+
+const findEllipsisElement = (element: HTMLElement): HTMLElement | null => {
+  const style = getComputedStyle(element);
+  if (style.textOverflow === 'ellipsis' || Number(style.webkitLineClamp) > 0) {
+    return element;
+  } else {
+    for (let i = 0; i < element.children.length; i++) {
+      const overflowElement = findOverflowElement(element.children[i] as HTMLElement);
+      if (overflowElement) {
+        return overflowElement;
+      }
+    }
+    return null;
+  }
+};
+
+const isOverflowed = (element: HTMLElement) => {
+  const overflowElement = findEllipsisElement(element) || findOverflowElement(element);
+
+  if (overflowElement) {
+    return (
+      overflowElement.scrollWidth > overflowElement.clientWidth ||
+      overflowElement.scrollHeight > overflowElement.clientHeight
+    );
+  }
+  return false;
+};
+
+export interface OverflowTooltipProps extends Omit<React.HTMLAttributes<HTMLDivElement>, 'title'> {
+  /**
+   * The target (anchor element) for the Tooltip.
+   *
+   * **Note:** This **must** be a single Element, StyledComponent, or any other component that
+   * forwards extra props to an Element. Tooltip works running `React.cloneElement` on the children
+   * and adds extra properties like aria attributes and event handlers. This is currently a
+   * limitation of the Tooltip component. Functionality will not work if this condition isn't met.
+   */
+  children: React.ReactElement<any>;
+  /**
+   * Sets the placement preference used by PopperJS.
+   * @default 'top'
+   */
+  placement?: Placement;
+}
+
+function mergeCallbacks<T extends {[key: string]: any}>(
```

Comment:
I assume this is something we can remove in v5, yeah?

Reply:
Yes. I'd use `mergeProps` from the compound component utility functions.

Reply:
Honestly, the way this works makes no sense to me and I wrote it!